src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -939,6 +939,29 @@ void uct_ib_mlx5_destroy_qp(uct_ib_mlx5_md_t *md, uct_ib_mlx5_qp_t *qp)
     }
 }
 
+size_t uct_ib_mlx5_devx_sq_length(size_t tx_qp_length)
+{
+    return ucs_roundup_pow2_or0(tx_qp_length * UCT_IB_MLX5_MAX_BB);
+}
+
+size_t uct_ib_mlx5_sq_length(const uct_ib_mlx5_md_t *md, size_t tx_qp_length,
+                             enum ibv_qp_type qp_type)
+{
+    switch (qp_type) {
+    case IBV_QPT_DRIVER:
+        if (md->flags & UCT_IB_MLX5_MD_FLAG_DEVX_DCI) {
+            return uct_ib_mlx5_devx_sq_length(tx_qp_length);
+        } else {
+            return ucs_roundup_pow2_or0(tx_qp_length * md->dv_tx_wqe_ratio.dc);
+        }
+    case IBV_QPT_UD:
+        return ucs_roundup_pow2_or0(tx_qp_length * md->dv_tx_wqe_ratio.ud);
+    default:
+        ucs_fatal("unexpected combinations of md_flags 0x%x and qp_type %d",
+                  md->flags, qp_type);
```

Comment:
error message is not fully correct now, because md flags can not lead to the failure anymore. It's good to print `md->flags`, but better rephrase error messages